File: iOS/lynx/Base/LynxUIUpdateDataAction.m
// Copyright 2017 The Lynx Authors. All rights reserved.

#import "LynxUIUpdateDataAction.h"

@implementation LynxUIUpdateDataAction

- (id) initWithTarget:(LynxRenderObjectImpl *)target andKey:(LynxRenderObjectAttr)key andData:(id)data {
    self = [super initWithType:DO_UPDATE_DATA_ACTION andTarget:target];
    if (self) {
        _key = key;
        _data = data;
    }
    return self;
}

- (void)doAction {
    [self.renderObjectImpl updateData:_data withKey:_key];
}

@end

File: Android/sdk/src/main/java/com/lynx/core/tree/LynxUIUpdateDataAction.java
// Copyright 2017 The Lynx Authors. All rights reserved.
package com.lynx.core.tree;

import com.lynx.core.impl.RenderObjectAttr;
import com.lynx.core.impl.RenderObjectImpl;

public class LynxUIUpdateDataAction extends LynxUIAction {
    private RenderObjectAttr mKey;
    private Object mParam;

    public LynxUIUpdateDataAction(RenderObjectImpl target,
                                  RenderObjectAttr key,
                                  Object param) {
        super(LynxUIActionType.DO_UPDATE_DATA_ACTION, target);
        mKey = key;
        mParam = param;
    }

    public RenderObjectAttr key() {
        return mKey;
    }

    @Override
    public void doAction() {
        mTarget.updateData(mKey, mParam);
    }
}
